src/core/api/thread_api.cpp

```diff
@@ -479,17 +479,17 @@ ThreadError otThreadGetParentInfo(otInstance *aInstance, otRouterInfo *aParentIn
     VerifyOrExit(aParentInfo != NULL, error = kThreadError_InvalidArgs);
 
     parent = aInstance->mThreadNetif.GetMle().GetParent();
-    memcpy(aParentInfo->mExtAddress.m8, parent->mMacAddr.m8, OT_EXT_ADDRESS_SIZE);
-
-    aParentInfo->mRloc16          = parent->mValid.mRloc16;
-    aParentInfo->mRouterId        = Mle::Mle::GetRouterId(parent->mValid.mRloc16);
-    aParentInfo->mNextHop         = parent->mNextHop;
-    aParentInfo->mPathCost        = parent->mCost;
-    aParentInfo->mLinkQualityIn   = parent->mLinkInfo.GetLinkQuality(aInstance->mThreadNetif.GetMac().GetNoiseFloor());
-    aParentInfo->mLinkQualityOut  = parent->mLinkQualityOut;
-    aParentInfo->mAge             = static_cast<uint8_t>(Timer::MsecToSec(Timer::GetNow() - parent->mLastHeard));
-    aParentInfo->mAllocated       = parent->mAllocated;
-    aParentInfo->mLinkEstablished = parent->mState == Neighbor::kStateValid;
+    memcpy(aParentInfo->mExtAddress.m8, &parent->GetExtAddress(), OT_EXT_ADDRESS_SIZE);
```

Comment:
Using OT_EXT_ADDRESS_SIZE here, but sizeof(parent->GetExtAddress()) elsewhere.
Should the code follow exactly one pattern?